tests/next_tests/unit_tests/test_common.py

```diff
@@ -64,3 +82,144 @@ def test_unit_range_index_error(rng):
 def test_unit_range_slice_error(rng):
     with pytest.raises(ValueError):
         rng[1:2:5]
+
+
+@pytest.mark.parametrize("rng1, rng2, expected", [
+    (UnitRange(0, 5), UnitRange(10, 15), UnitRange(0, 0)),
+    (UnitRange(0, 5), UnitRange(5, 10), UnitRange(5, 5)),
+    (UnitRange(0, 5), UnitRange(3, 7), UnitRange(3, 5)),
+    (UnitRange(0, 5), UnitRange(1, 6), UnitRange(1, 5)),
+    (UnitRange(0, 5), UnitRange(-5, 5), UnitRange(0, 5)),
+    (UnitRange(0, 0), UnitRange(0, 5), UnitRange(0, 0)),
+    (UnitRange(0, 0), UnitRange(0, 0), UnitRange(0, 0)),
+])
+def test_unit_range_intersection(rng1, rng2, expected):
+    result = rng1 & rng2
+    assert result == expected
+
+
+def test_domain_length(domain):
+    assert len(domain) == 3
+
+
+def test_domain_contains_range(domain):
+    assert UnitRange(0, 10) in domain
+    assert UnitRange(5, 15) in domain
+    assert UnitRange(20, 30) in domain
+    assert UnitRange(25, 28) in domain
+
+
+def test_domain_contains_element(domain):
+    assert 0 in domain
+    assert 5 in domain
+    assert 20 in domain
+    assert 9 in domain
+    assert 14 in domain
+
+
+def test_domain_not_contains_range(domain):
+    assert UnitRange(25, 35) not in domain
+    assert UnitRange(10, 16) not in domain
+
+
+def test_domain_not_contains_element(domain):
+    assert 100 not in domain
+    assert -5 not in domain
+    assert 16 not in domain
+
+
+def test_positive_infinity_range():
+    pos_inf_range = UnitRange(Infinity.positive(), Infinity.positive())
+    assert len(pos_inf_range) == 0
+
```

Comment:
please add a test that intersection with Infinity works correctly (because if we change Infinity later, its important we don't change the behavior of UnitRange)